### PR TITLE
Subscriptions: Add flag to not send email if updating published post

### PR DESCRIPTION
**Pull request by ebinnion**
It's been reported recently that when a user updates a post that subscription emails can be sent out for that post.

In my testing, I was able to reproduce when I deleted the synced post on WPCOM along with its meta. So, if the meta that signaled we had sent a post was missing for some reason on the WPCOM side, we would send that subscription again.

This changeset ensures that we set the `_jetpack_dont_email_post_to_subs` meta whenever we are updating an already published post.

To test:

- Checkout `update/dont-send-sub-email-old-update-post` branch
- Publish a post on your site and ensure subscription email goes out
- Delete the post on WPCOM
- Update post
- Ensure no subscription email goes out

cc @lezama @gravityrail for code review

### CODE REVIEW

**Comment by gravityrail** [comment]
I love it. Tests?

File: readme.txt [hunk 1]
```diff
@@ -89,9 +89,10 @@ There are opportunities for developers at all levels to contribute. [Learn more
 * Add deprecated PHP file back, to avoid generating Fatal Errors on sites with aggressive caching. #5098
 * Make sure concatenated CSS is generated for RTL languages. #5095
 * To improve sync performance, add snapTW to the list of post meta data that won't be synchronized for each post. That meta data can be 45mb+ per post. #5092
-* Admin Page: make sure all translated strings are encoded properly. #5101
-* Admin Page: only use POST requests for updating the state of Jetpack, to avoid issues on servers not allowing PUT requests. #5100
-
+* Admin Page: make sure all translated strings are encoded properly. #5101
```
Comment (ebinnion) [inline readme.txt]: I'm not sure why these other lines are showing up in the changeset, they look identical to me. Any thoughts?

Comment (jeherve) [inline readme.txt]: Yeah, that's weird. I don't notice any differences.

**Comment by ebinnion** [comment]
I removed the changelog commit from this PR.

**Pull request merged**